### PR TITLE
Add generic to frozen array

### PR DESCRIPTION
**Pull request by gterzian**
https://github.com/servo/servo/pull/25467#discussion_r364575071

Depends on https://github.com/servo/servo/pull/25467

---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

**Comment by gterzian** [comment]
@Manishearth r?

File: components/script/dom/messageevent.rs [hunk 1]
```diff
@@ -219,7 +238,13 @@ impl MessageEventMethods for MessageEvent {
 
     /// <https://html.spec.whatwg.org/multipage/#dom-messageevent-ports>
     fn Ports(&self, cx: JSContext) -> JSVal {
-        message_ports_to_frozen_array(self.ports.borrow().as_slice(), cx)
+        let ports: Vec<DomRoot<MessagePort>> = self
```
Comment (Manishearth) [inline components/script/dom/messageevent.rs]: It's actually worth caching this in a `MutNullableDom<Heap>` or something. The spec doesn't say it's SameObject, but it probably should, and even if it doesn't we can still do it.

**Review by Manishearth — APPROVED** [review]

File: components/script/dom/messageevent.rs (hunk 1 above)
Comment (Manishearth) [reply components/script/dom/messageevent.rs]: Doesn't have to be this PR

Comment (gterzian) [reply components/script/dom/messageevent.rs]: Ok, I will look into it.

**Comment by gterzian** [comment]
@Manishearth r?

**Comment by Manishearth** [comment]
```
1 unexpected results that are NOT known-intermittents:
  ▶ Unexpected subtest result in /html/webappapis/scripting/events/messageevent-constructor.https.html:
  └ PASS [expected FAIL] ports attribute should be a FrozenArray
```

**Comment by gterzian** [comment]
`assert_true(ev.ports === ev.ports, "ev.ports should return the same object")`, so the caching was  necessary afterall...

**Pull request merged**